horust/src/horust/healthcheck/checks.rs

```diff
@@ -68,3 +80,57 @@ impl Check for FilePathCheck {
             .unwrap_or(Ok(()))
     }
 }
+
+pub(crate) struct CommandCheck {}
+
+impl CommandCheck {
+    fn prepare_cmd(&self, cmd: &str) -> anyhow::Result<()> {
+        let mut chunks = shlex::split(cmd).context(format!("Failed to split command: {}", cmd))?;
+        let program = chunks
+            .first()
+            .context(format!("Failed to get program from command: {}", cmd))?;
+        let path = if program.contains('/') {
+            program.to_string()
+        } else {
+            find_program(program)?
+        };
+        chunks[0] = path;
+        get_parsed_commands()
+            .lock()
+            .unwrap()
+            .insert(cmd.to_string(), chunks);
+        Ok(())
+    }
+}
+
+impl Check for CommandCheck {
+    fn run(&self, healthiness: &Healthiness) -> bool {
+        healthiness
+            .command
+            .as_ref()
+            .map(|command| {
+                let parsed_command = get_parsed_commands().lock().unwrap().get(command).cloned();
```

Comment:
would it work without the cloned? Or won't because of the unwrap? Because I can see later we just use a reference to the command's parts

Reply:
`cloned` is used because of the Mutex. Otherwise, I have to hold the lock until it finishes the Command which might take too long for a lock.

Reply:
Got it, makes sense thank you!